pkg/networkserver/downlink.go

```diff
@@ -112,6 +98,45 @@ func (s generateDownlinkState) appendApplicationUplinks(ups []*ttnpb.Application
 	}
 }
 
+// deviceTransmissionDelay returns the estimated one-way transmission delay between
+// frame being trasmitted by the device and received by the Network Server and other way round.
+func deviceTransmissionDelay(dev *ttnpb.EndDevice, defaults ttnpb.MACSettings, phy band.Band) time.Duration {
+	if dev.MACSettings != nil && dev.MACSettings.DesiredRx1Delay != nil {
+		return dev.MACSettings.DesiredRx1Delay.Value.Duration() / 2
+	} else if dev.PendingMACState != nil {
+		return dev.PendingMACState.DesiredParameters.Rx1Delay.Duration() / 2
+	} else if dev.MACState != nil {
+		return dev.MACState.DesiredParameters.Rx1Delay.Duration() / 2
```

Comment:
Why does desired take precedence?

For style; afaik this gives lint warnings (else-if after return); consider switch cases

Reply:
Because *desired* will always contain the most up-to-date information.

Reply:
But if my RX1 delay is 1 second, and I'm changing it to 15 seconds, will class A downlink ever make it? If so, how?

Reply:
Changed this to always using the active Rx1Delay except for join-requests